### DISCUSSION TITLE
Change Author

### DISCUSSION THREAD

**Discussion by PhotoKevin** [mathieucarbou/ESPAsyncWebServer#129] [General]
The ESPAsyncWebServer shows up in the Arduino IDE as being by Me-No-Dev. This is confusing since a user will look at that and think they should go to the Me-No-Dev version on github for issues.
![image](https://github.com/user-attachments/assets/52d4c89c-d195-4c93-9615-74da20879c76)

See the issue there as [https://github.com/me-no-dev/ESPAsyncWebServer/issues/1437](https://github.com/me-no-dev/ESPAsyncWebServer/issues/1437). 

I certainly see why you wouldn't want to claim ownership of the original code, but there should be a way to make it obvious in the IDE that this is not the Me-No-Dev version. Possibly the author field in library.properties could be change to "Mathiew Carbou (via Me-No-Dev)" or something? (I don't know the syntax limitations on the author field, this may not be possible). And yes, I see that the maintainer field is filled out correctly, but that doesn't show in the IDE and you have to know what you're doing to find it.

**Comment by mathieucarbou** [comment]
Copy of my answer posted here: https://github.com/me-no-dev/ESPAsyncWebServer/issues/1437#issuecomment-2408847528

-----

> I certainly see why you wouldn't want to claim ownership of the original code,

That's the opposite. Please read:

- https://forum.arduino.cc/t/ide-library-manager-handling-of-duplicate-library-names/1306491/14
- https://github.com/arduino/library-registry/pull/4585

Note: I've already raised myself some issues regarding how libraries are handled within Arduino Registry.
IMO, it's a mess. They are not handling dependencies correctly. 

For what is worth, and Arduino Registry guys agree with me: **author field should not ne changed, maintainer field has to be added, attribution has to remain.**

**As a reminder, mathieucarbou/ESPAsyncWebServer is a maintained fork by me and some other people**

Also, I would say: stop using toy IDEs like Arduino IDE.  You have real development IDE already available like Vscode and the **pioarduino** extension which provides a good and more reliable dependency management.

The day Arduino IDE will use a correct dependency management system, my opinion will change, but for now, the team behind it does not focus on developers writing code.

**Comment by PhotoKevin** [comment]
If that's their rule, that's it. 

As to the IDE suggestion; Noted and rejected.
